### ISSUE TITLE
Cloud Messaging subscription error

### ISSUE THREAD

**Issue by eitanaviv** [firebase/firebase-ios-sdk#2625]
### Describe your environment

  * Xcode version: 10.2 beta 4 (Swift 4.2 and latest iOS 12.2 beta 6)
  * Firebase SDK version: 5.19.0
  * Firebase Component: Messaging
  * Component version: 3.4.0

### Describe the problem
I'm trying to subscribe the user to a topic named after his phone number (only numbers) and at installation it works perfectly, but after running the app once again it fails to subscribe him into that topic and he don't get notifications anymore.

This is the error i'm getting:
`The operation couldn’t be completed. (com.google.fcm error 0.)`

And another error i'm getting is:
`[Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"`

#### Steps to reproduce:

Implement the basic code of Cloud Messaging.

#### Relevant Code:

AppDelegate:
```
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        FirebaseApp.configure()
        UIApplication.shared.registerForRemoteNotifications()

        return true
    }


    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any]) {
        print(userInfo)
    }
    
    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any],
                     fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
        print(userInfo)
        
        completionHandler(UIBackgroundFetchResult.newData)
    }
```

Main ViewController (it belongs to a different storyboard after logging in to the app):
```
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        let center = UNUserNotificationCenter.current()
        let authOptions: UNAuthorizationOptions = [.alert, .badge, .sound]
        center.requestAuthorization(options: authOptions) { granted, error in
            InstanceID.instanceID().instanceID { (result, error) in
                if let error = error {
                    print("Error fetching remote instance ID: \(error)")
                } else {
                    center.delegate = self
                }
            }
        }
        
        if let userPhone = UserDefaults.standard.string(forKey: "userPhone") {
            Messaging.messaging().subscribe(toTopic: userPhone) { error in
                if error != nil {
                    print("Subscription error: \(error?.localizedDescription ?? "")")
                } else {
                    print("Subscribed to topic \(userPhone)")
                }
            }
        }
      }
```

**Comment by chliangGoogle** [comment]
Is your FCM token changed the second time app starts? I'm wondering if it's a similar issue like #2438

**Comment by stshelton** [comment]
I am receiving the same issue as described above. On the first run, everything works correctly but on the second run, I receive the same errors described above. I'm using firebase version 5.20.2 with firebase messaging version 3.5.0 and have been reading issue #2438 , from my error logs I don't see FCM token changing on the second run of the app. It also seems when I receive these "failed to subscribe errors" my push notifications stop working altogether also. I provided an error log with debug logging enabled.

Error log
2019-04-26 14:21:21.203700-0400 HummingBiird_App_Dev[1292:278618]  - <AppMeasurement>[I-ACS036002] Analytics screen reporting is enabled. Call +[FIRAnalytics setScreenName:setScreenClass:] to set the screen name or override the default screen class name. To disable screen reporting, set the flag FirebaseScreenReportingEnabled to NO (boolean) in the Info.plist
2019-04-26 14:21:21.326217-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Analytics][I-ACS024000] Debug mode is on
2019-04-26 14:21:21.326301-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Analytics][I-ACS023007] Analytics v.50801000 started
2019-04-26 14:21:21.326376-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Analytics][I-ACS023008] To disable debug logging set the following application argument: -noFIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2019-04-26 14:21:21.326555-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Analytics][I-ACS023009] Debug logging enabled
2019-04-26 14:21:21.344725-0400 HummingBiird_App_Dev[1292:278616] 5.20.0 - [Firebase/Analytics][I-ACS029014] Successfully parsed a configuration. Version: 1541806582101691
2019-04-26 14:21:21.345245-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/Analytics][I-ACS023016] Analytics is ready to receive events
2019-04-26 14:21:21.346647-0400 HummingBiird_App_Dev[1292:278616] 5.20.0 - [Firebase/Analytics][I-ACS023080] Setting user property. Name, value: first_open_after_install (_fi), 1
2019-04-26 14:21:21.348269-0400 HummingBiird_App_Dev[1292:278616] 5.20.0 - [Firebase/Analytics][I-ACS023087] User property set. Name, value: first_open_after_install (_fi), 1
2019-04-26 14:21:21.361167-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
- [Firebase/Analytics][I-ACS023024] No data to upload. Upload task will not be scheduled
2019-04-26 14:21:21.361261-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS023012] Analytics enabled
2019-04-26 14:21:21.367914-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 14:21:21.375874-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 14:21:21.376561-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 14:21:21.376712-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/InstanceID][I-IID009000] No keypair info is found with tag com.google.iid-|K|
2019-04-26 14:21:21.377748-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/InstanceID][I-IID009006] There's no legacy keypair so no need to do migration.
2019-04-26 14:21:21.379466-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording trace: _p
2019-04-26 14:21:21.383661-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Messaging][I-FCM002000] FIRMessaging library version 3.5.0
2019-04-26 14:21:21.385726-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Messaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
2019-04-26 14:21:21.510384-0400 HummingBiird_App_Dev[1292:278619] 5.20.0 - [Firebase/Firestore][I-FST000001] Initializing. Current user: fYIiMLXwLGeCHqESJhr22iGH8WY2
2019-04-26 14:21:21.517424-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) start
2019-04-26 14:21:21.517628-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Auth][I-AUT000002] Token auto-refresh enabled.
2019-04-26 14:21:21.517976-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Auth][I-AUT000004] Token auto-refresh scheduled in 28:11 for the new token.
2019-04-26 14:21:21.580407-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS031010] Tracking view controller. Class, ID: HomeViewController, 1542226693968786025
2019-04-26 14:21:21.580482-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS031013] Screen view event not logged. App is not active.
2019-04-26 14:21:21.581115-0400 HummingBiird_App_Dev[1292:278647] 5.20.0 - [Firebase/Analytics][I-ACS031006] View controller already tracked. Class, ID: HomeViewController, 1542226693968786025
2019-04-26 14:21:21.581222-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS031006] View controller already tracked. Class, ID: HomeViewController, 1542226693968786025
2019-04-26 14:21:21.582722-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] Credential Changed. Current user: fYIiMLXwLGeCHqESJhr22iGH8WY2
2019-04-26 14:21:21.582909-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] RemoteStore 282096000 restarting streams for new credential
2019-04-26 14:21:21.583128-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) stop
2019-04-26 14:21:21.583196-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) stop
2019-04-26 14:21:21.583333-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) start
2019-04-26 14:21:21.601822-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) start
2019-04-26 14:21:21.601908-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/Firestore][I-FST000001] Creating Firestore stub.
2019-04-26 14:21:21.601953-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/Firestore][I-FST000001] roots.pem found in bundle org.cocoapods.gRPCCertificates (in parent org.cocoapods.grpcpp)
2019-04-26 14:21:21.605401-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS033003] Scheduling user engagement timer
2019-04-26 14:21:21.605529-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS002002] Engagement timer scheduled to fire in approx. (s): 3600
2019-04-26 14:21:21.608053-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording trace: _as
2019-04-26 14:21:21.610598-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) watch: <GCFSListenRequest 0x2837af800>: {
2019-04-26 14:21:21.611092-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStr2019-04-26 14:21:21.612394-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) watch: <GCFSListenRequest 0x283791b20>: {
pp_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) watch: <GCFSListenRequest 0x2837915e0>: {
2019-04-26 14:21:21.612827-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097912019-04-26 14:21:21.612902-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) initial request: <GCFSWriteRequest 0x283285900>: {
2019-04-26 14:21:21.614034-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS023051] Logging event: origin, name, params: auto, screen_view (_vs), {
    firebase_event_origin (_o) = auto    firebase_screen_class (_sc) = HomeViewController;
2019-04-26 14:21:21.615659-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS023073] Debug mode is enabled. Marking event as    firebase_debug (_dbg) = 1;
    firebase_event_origin (_o) = auto;
    firebase_screen_class (_sc) = HomeViewController;
2019-04-26 14:21:21.624900-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS031010] Tracking view controller. Class, ID: SPPermissionDialogController, 1542226693968786026
2019-04-26 14:21:21.629959-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS023072] Event logged. Event name, event params: screen_view (_vs), {
    firebase_debug (_dbg) = 1;
    firebase_event_origin (_o) = auto;
    firebase_screen_class (_sc) = HomeViewController;
2019-04-26 14:21:21.718006-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS002002] Measurement timer scheduled to fire in approx. (s): 0.8878859281539917
2019-04-26 14:21:21.718090-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS023028] Upload task scheduled to be executed in approx. (s): 0.8878859281539917
2019-04-26 14:21:21.718726-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS023051] Logging event: origin, name, params: auto, screen_view (_vs), {
    firebase_event_origin (_o) = auto;
    firebase_previous_id (_pi) = 1542226693968786025;
    firebase_screen_class (_sc) = SPPermissionDialogController;
2019-04-26 14:21:21.719570-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS023073] Debug mode is enabled. Marking event as debug and real-time. Event name, parameters: screen_view (_vs), {
    firebase_debug (_dbg) = 1;
    fire    firebase_previous_id (_pi) = 1542226693968786025;
    firebase_screen_class (_sc) = SPPermissionDialogController;
2019-04-26 14:21:21.725147-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Analytics][I-ACS023072] Event logged. Event name, event params: screen_view (_vs), {
    firebase_debug (_dbg) = 1;
    firebase_event_origin (_o) = auto;
    firebase_    firebase_screen_class (_sc) = SPPermissionDialogController;
2019-04-26 14:21:21.811695-0400 HummingBiird_App_Dev[1292:278664] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://data.flurry.com/aas.do
2019-04-26 14:21:21.812163-0400 HummingBiird_App_Dev[1292:278664] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://data.flurry.com/aas.do
2019-04-26 14:21:21.819465-0400 HummingBiird_App_Dev[1292:278618] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://e.crashlytics.com/spi/v2/events
2019-04-26 14:21:21.828403-0400 HummingBiird_App_Dev[1292:278630] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://data.flurry.com/aas.do
2019-04-26 14:21:21.841264-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Analytics][I-ACS023027] Do not schedule an upload task. Task already exists. Will be executed in seconds: 0.7644479274749756
2019-04-26 14:21:22.133669-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://graph.facebook.com/v3.2/228883907832743/activities
2019-04-26 14:21:22.263128-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://cdn.branch.io/sdk/uriskiplist_v2.json
2019-04-26 14:21:22.309316-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.310843-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) response: <GCFSWriteResponse 0x2837a52d0>: {
2019-04-26 14:21:22.313932-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) write request: <GCFSWriteRequest 0x2832afe80>: {
2019-04-26 14:21:22.317666-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) write request: <GCFSWriteRequest 0x283288080>: {
2019-04-26 14:21:22.412156-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.413642-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283296000>: {
2019-04-26 14:21:22.415330-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.415580-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283285380>: {
2019-04-26 14:21:22.415739-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283285400>: {
Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.417035-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] Wat2019-04-26 14:21:22.417136-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST00002019-04-26 14:21:22.417653-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283287100>: {
2019-04-26 14:21:22.491678-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.492975-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) response: <GCFSWriteResponse 0x28378cfc0>: {
2019-04-26 14:21:22.498709-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.499567-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283295900>: {
2019-04-26 14:21:22.500916-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.501163-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x28328fd80>: {
2019-04-26 14:21:22.503796-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.506476-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283287d00>: {
2019-04-26 14:21:22.511703-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2.513046-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283295280>: {
2019-04-26 14:21:22.513546-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelist2019-04-26 14:21:22.516235-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283295300>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/fYIiMLXwLGeCHqESJhr22iGH8WY2?alt=media&token=4797ce75-d841-44d8-8b22-9e788919a22a"
2019-04-26 14:21:22.518090-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] Wat2019-04-26 14:21:22.518243-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283289300>[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.519250-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchS2019-04-26 14:21:22.521745-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Fire2019-04-26 14:21:22.522588-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSList2019-04-26 14:21:22.523044-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283290c80>: {
2019-04-26 14:21:22.528702-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.528825-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCF2019-04-26 14:21:22.528908-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
ngBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283290c00>: {
2019-04-26 14:21:22.529057-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStre2019-04-26 14:21:22.529426-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832a5f00>: {
21:22.529704-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
irebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x283295180>: {
2019-04-26 14:21:22.530918-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: F2019-04-26 14:21:22.532244-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832add00>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Fcleaning_bedroom%402x.png?alt=media&token=30292605-374f-41f5-bce4-4aa0ed718a0f"
939-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.533561-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/F2019-04-26 14:21:22.533937-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 20192019-04-26 14:21:22.534483-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832aea80>: {
2019-04-26 14:21:22.534772-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2.535966-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832ad400>: {
 "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Fwriting_generalhelp%402x.png?alt=media&token=c89bfcca-3c53-44c3-8a36-402a2d8b0493"
[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.537915-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchS            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/Su2019-04-26 14:21:22.538313-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelis2019-04-26 14:21:22.539304-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4400>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Ffitness_dietplan%402x.png?alt=media&token=a1f3584c-03df-4cf6-9607-3c2c8fb7963f"
9734-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.540167-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4100>: {
2019-04-26 14:21:22.540387-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2012019-04-26 14:21:22.541298-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4500>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Ffitness_personaltraining%402x.png?alt=media&token=c31fb4d0-411b-4c86-b43a-ebf961e6445a"
400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.542182-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firest 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.543306-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Ffitness2019-04-26 14:21:22.544030-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:2019-04-26 14:21:22.544484-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4880>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/SubCategory%20Images%2Fwriting_generalhelp%402x.png?alt=media&token=c89bfcca-3c53-44c3-8a36-402a2d8b0493"
4893-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.545532-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4980>: {
            string_value: "https://firebasestorage.googleapis.com/v0/b/hummingbiird2019-04-26 14:21:22.546151-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (28202019-04-26 14:21:22.546304-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4a80>: {
            string_value: "https://firebasestorage.googleapis2019-04-26 14:21:22.546630-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
92:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4b80>: {
            string_value: "https://firebasestorage.go2019-04-26 14:21:22.547508-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Fire2019-04-26 14:21:22.547595-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSList00 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.547848-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firesto2019-04-26 14:21:22.547955-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 G2019-04-26 14:21:22.548007-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e4d80>: {
4:21:22.569961-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) unwatch: <GCFSListenRequest 0x2837f1ce0>: {
019-04-26 14:21:22.570197-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e6800>: {
2019-04-26 14:21:22.570806-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-F2019-04-26 14:21:22.570868-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse2019-04-26 14:21:22.618849-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.619193-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Firestore][I-FST000001] WriteStream (10a180b78) response: <GCFSWriteResponse 0x2837ffb80>: {
2019-04-26 14:21:22.620472-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS002001] Measurement timer fired
2019-04-26 14:21:22.620518-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS002003] Measurement timer canceled
2019-04-26 14:21:22.620546-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS023033] Starting data upload
2019-04-26 14:21:22.622958-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Analytics][I-ACS023105] Event is not subject to real-time event count daily limit. Marking an event as real-time. Event name, parameters: screen_view (_vs), {
    firebas    firebase_event_origin (_o) = auto;
    firebase_screen_class (_sc) = HomeViewController;
2019-04-26 14:21:22.624581-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.624807-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e7480>: {
2019-04-26 14:21:22.630080-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS023105] Event is not subject to real-time event count daily limit. Marking an event as real-time. Event name, parameters: screen_view (_vs), {
    firebase_debug (_dbg) = 1;
    firebase_event_origin (_o) = auto;
    firebase_previous_id (_pi) = 1542226693968786025;
    firebase_screen_class (_sc) = SPPermissionDialogController;
2019-04-26 14:21:22.631884-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Analytics][I-ACS012018] Saving bundle. size (bytes): 507
2019-04-26 14:21:22.632043-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Analytics][I-ACS023116] Bundle added to the upload queue. BundleID, timestamp (ms): 11, 1556302881621
2019-04-26 14:21:22.635171-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Analytics][I-ACS023038] Uploading events. Elapsed time since last successful upload (s): 872.1074950695038
2019-04-26 14:21:22.635415-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Analytics][I-ACS023039] Measurement data sent to network. Timestamp (ms), data: 1556302882634, <APMPBMeasurementBatch: 0x281e9c5b0>
2019-04-26 14:21:22.639075-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Analytics][I-ACS900000] Uploading data. Host: https://app-measurement.com/a
}, "profile_pic": https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/fYIiMLXwLGeCHqESJhr22iGH8WY2?alt=media&token=4797ce75-d841-44d8-8b22-9e788919a22a, "first_name": Dwyane, "email": wade@wade.com]
["profile_pic": https://firebasestorage.googleapis.com/v0/b/hummingbiird-dev.appspot.com/o/fYIiMLXwLGeCHqESJhr22iGH8WY2?alt=media&token=4797ce75-d841-44d8-8b22-9e788919a22a, "ratings": {
2019-04-26 14:21:22.663842-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) watch: <GCFSListenRequest2019-04-26 14:21:22.665366-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) watch: <GCFSListenRequest 0x283794d20>: {
21:22.665769-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) unwatch: <GCFSListenRequest 0x2837f2610>: {
-04-26 14:21:22.665916-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) unwatch: <GCFSListenRequest 0x2837f2680>: {
2019-04-26 14:21:22.731425-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Analytics][I-ACS901006] Received SSL challenge for host. Host: https://app-measurement.com/a
2019-04-26 14:21:22.740836-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.741129-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2a00>: {
2019-04-26 14:21:22.741369-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.741415-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2b00>: {
2019-04-26 14:21:22.741561-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.741596-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2b80>: {
2019-04-26 14:21:22.741628-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.741762-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2c00>: {
2019-04-26 14:21:22.807527-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.808041-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832f6580>: {
 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.808587-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) re14:21:22.808673-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
 [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e3200>: {
2019-04-26 14:21:22.812204-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
:21:22.812642-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e3280>: {
2019-04-26 14:21:22.814386-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
-26 14:21:22.814479-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2a80>: {
2019-04-26 14:21:22.814537-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) header2019-04-26 14:21:22.814577-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e3300>: {
2019-04-26 14:21:22.841300-0400 HummingBiird_App_Dev[1292:278661] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://app-measurement.com/a
2019-04-26 14:21:22.843897-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Analytics][I-ACS023044] Successful upload. Got network response. Code, size: 204, -1
2019-04-26 14:21:22.846160-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Analytics][I-ACS002002] Measurement timer scheduled to fire in approx. (s): -0.2246079444885254
2019-04-26 14:21:22.846517-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Analytics][I-ACS023028] Upload task scheduled to be executed in approx. (s): -0.2246079444885254
2019-04-26 14:21:22.847462-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.847591-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e1f80>: {
[Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.847865-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) respo2019-04-26 14:21:22.853104-0400 HummingBiird_App_Dev[1292:278646] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) unwatch: <GCFSListenRequest 0x2837f2680>: {
2019-04-26 14:21:22.854296-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=fxg9HTNNvEU:APA91bFXCtRjCz1Bdjua9SI9lPQ1AmfWVRVuBPlf2dVX6btKqvbbnLgpz0s90hDwWv0azGEfqpC1NMm8lKQPsdRpOLngjHDHYEp81292:278650] 5.20.0 - [Firebase/Analytics][I-ACS023024] No data to upload. Upload task will not be scheduled
2019-04-26 14:21:22.854408-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Analytics][I-ACS002003] Measurement timer canceled
6 14:21:22.855126-0400 HummingBiird_App_Dev[1292:278609] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=fxg9HTNNvEU:APA91bFXCtRjCz1Bdjua9SI9lPQ1AmfWVRVuBPlf2dVX6btKqvbbnLgpz0s90hDwWv0azGEfqpC1NMm8lKQPsdRpOLngjHDHYEp8eGmm4bgx- - [Firebase/Firestore][I-FST000001] WatchStream (282097918) unwatch: <GCFSListenRequest 0x2837f25a0>: {
2019-04-26 14:21:22.881003-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.881189-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832ee600>: {
2019-04-26 14:21:22.886350-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) headers (whitelisted): date: Fri, 26 Apr 2019 18:21:22 GMT
2019-04-26 14:21:22.886477-0400 HummingBiird_App_Dev[1292:278627] 5.20.0 - [Firebase/Firestore][I-FST000001] WatchStream (282097918) response: <GCFSListenResponse 0x2832e2880>: {
2019-04-26 14:21:22.997134-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-26 14:21:22.997859-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://iid.googleapis.com/iid/register
2019-04-26 14:21:23.014575-0400 HummingBiird_App_Dev[1292:278650] 5.20.0 - **[Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"**
2019-04-26 14:21:23.028030-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-26 14:21:23.028618-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - **[Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"**
2019-04-26 14:21:23.028708-0400 HummingBiird_App_Dev[1292:278617] 5.20.0 - [Firebase/Performance][I-PRF100009] Recording network trace: https://iid.googleapis.com/iid/register
`

**Comment by chliangGoogle** [comment]
@stshelton From your log, you are not sending me the full context, like "topic subscription request". Also some of the logs seem missing. Can you send the full log of anything InstanceID and Messaging Log related so I can get a better idea on why the request was failing? Thank you!

**Comment by stshelton** [comment]
I searched instanceID and Messaging within debug window. If you want all of debug window I can copy and paste it, it just logged a lot of info that I feel like u don't need. I was trying to save all of log but i couldnt find a way.

**instanceID:**
2019-04-26 16:22:26.888478-0400 HummingBiird_App_Dev[1687:320114] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 16:22:26.894300-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 16:22:26.894400-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/InstanceID][I-IID023000] Info is not found in Keychain. OSStatus: -25300. Keychain query: {
2019-04-26 16:22:26.894473-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/InstanceID][I-IID009000] No keypair info is found with tag com.google.iid-|K|
13-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/InstanceID][I-IID009006] There's no legacy keypair so no need to do migration.

**messaging**
2019-04-26 16:22:26.919415-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/Messaging][I-FCM002000] FIRMessaging library version 3.5.0
2019-04-26 16:22:26.919876-0400 HummingBiird_App_Dev[1687:320118] 5.20.0 - [Firebase/Messaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
2019-04-26 16:22:28.110298-0400 HummingBiird_App_Dev[1687:320133] 5.20.0 - [Firebase/Messaging][I-FCM012019-04-26 16:22:28.110626-0400 HummingBiird_App_Dev[1687:320133] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=fxg9HTNNvEU:2019-04-26 16:22:28.250350-0400 HummingBiird_App_Dev[1687:320123] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-26 16:22:28.250630-0400 HummingBiird_App_Dev[1687:320123] 5.20.0 - [Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"
2019-04-26 16:22:28.252865-0400 HummingBiird_App_Dev[1687:320121] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-26 16:22:28.252944-0400 HummingBiird_App_Dev[1687:320121] 5.20.0 - [Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"

**Comment by stshelton** [comment]
It seems if I downgrade to firebase version 5.15.0. Push notifications/topics are working as intended, but only if I delete app and redownload it. Which there are some live users that have this version of the app and I would like to be able to fix push notifications with an update that doesn't require the user to delete app and redownload.

**Comment by pupprdev** [comment]
I'm also experiencing the same issue. Hope the following info is helpful:

## Describe your environment
Xcode version: 10.2.1 (10E1001)
Firebase SDK version: 5.20.2
Firebase Component: Messaging
Component version: 3.5.0

## Debug logs
Instead of pasting my full logs, I'm only pasting what I believe to be the relevant parts.

### First app start (v5.20.2)
```
2019-04-28 01:07:52.243640-0700 Puppr-DEV[22812:10048118] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=cngDGp5-piY:APA91bEajzOlWdYdbtZODQ_ixQGmJj5m_hnoAgbA58A2qZcKf8J_A2LNnQ3xQM0twgwJ8pF35iddCPPXkuX3ePbDYrOFGcF1yWYxfNuUlu0SGbkiwf7HmR11Q_LGvLWbyA8BSeiHRu9D&app=com.chinandcheeks.dogtrainer&device=5669145430566317485&app_ver=0.1&X-gcm.topic=/topics/timedSales&X-scope=/topics/timedSales

2019-04-28 01:07:52.932493-0700 Puppr-DEV[22812:10048126] 5.20.0 - [Firebase/Messaging][I-FCM004003] Successfully subscribed to topic /topics/timedSales
```

### Second app start (v5.20.2)
```
2019-04-28 01:08:58.411008-0700 Puppr-DEV[22817:10048791] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=cngDGp5-piY:APA91bEajzOlWdYdbtZODQ_ixQGmJj5m_hnoAgbA58A2qZcKf8J_A2LNnQ3xQM0twgwJ8pF35iddCPPXkuX3ePbDYrOFGcF1yWYxfNuUlu0SGbkiwf7HmR11Q_LGvLWbyA8BSeiHRu9D&app=com.chinandcheeks.dogtrainer&device=4851790658297917259&app_ver=0.1&X-gcm.topic=/topics/timedSales&X-scope=/topics/timedSales

2019-04-28 01:08:58.606766-0700 Puppr-DEV[22817:10048789] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-28 01:08:58.606881-0700 Puppr-DEV[22817:10048789] 5.20.0 - [Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"
```

### Third app start (v5.20.2)
```
2019-04-28 01:10:19.412870-0700 Puppr-DEV[22821:10049215] 5.20.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=cngDGp5-piY:APA91bEajzOlWdYdbtZODQ_ixQGmJj5m_hnoAgbA58A2qZcKf8J_A2LNnQ3xQM0twgwJ8pF35iddCPPXkuX3ePbDYrOFGcF1yWYxfNuUlu0SGbkiwf7HmR11Q_LGvLWbyA8BSeiHRu9D&app=com.chinandcheeks.dogtrainer&device=5372838138733384695&app_ver=0.1&X-gcm.topic=/topics/timedSales&X-scope=/topics/timedSales

2019-04-28 01:10:19.612324-0700 Puppr-DEV[22817:10048789] 5.20.0 - [Firebase/Messaging][I-FCM017002] Invalid registration response Error=INVALID_PARAMETERS
2019-04-28 01:10:19.613271-0700 Puppr-DEV[22817:10048789] 5.20.0 - [Firebase/Messaging][I-FCM004001] Failed to subscribe to topic Error Domain=com.google.fcm Code=0 "(null)"
```

**Notice that the `device` param value changes with every app start.** 

---

Here's what my debug logs look like after downgrading to Firebase SDK version 5.15.0:

### First app start (v5.15.0)
```
2019-04-28 01:32:31.005339-0700 Puppr-DEV[22848:10055628] 5.15.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=eo6vuaxmLcU:APA91bHHHlomRbZz-L7x7pITgzAJfzH30Ctfwf3WDOJWPqEDIzVbxbTiINA7myYjSUrR7CQfK_IZSxtLhpDffM-xra6O3WBLCqY94JTBaG704nqTZOdqlCsJaWOVxWE2-ickHqGEdRSO&app=com.chinandcheeks.dogtrainer&device=5280035528981645835&app_ver=0.1&X-gcm.topic=/topics/timedSales&X-scope=/topics/timedSales

2019-04-28 01:32:31.784553-0700 Puppr-DEV[22848:10055627] 5.15.0 - [Firebase/Messaging][I-FCM004003] Successfully subscribed to topic /topics/timedSales
```

### Second app start (v5.15.0)
```
2019-04-28 01:33:14.106642-0700 Puppr-DEV[22856:10056240] 5.15.0 - [Firebase/Messaging][I-FCM017000] Topic subscription request: sender=eo6vuaxmLcU:APA91bHHHlomRbZz-L7x7pITgzAJfzH30Ctfwf3WDOJWPqEDIzVbxbTiINA7myYjSUrR7CQfK_IZSxtLhpDffM-xra6O3WBLCqY94JTBaG704nqTZOdqlCsJaWOVxWE2-ickHqGEdRSO&app=com.chinandcheeks.dogtrainer&device=5280035528981645835&app_ver=0.1&X-gcm.topic=/topics/timedSales&X-scope=/topics/timedSales

2019-04-28 01:33:14.888842-0700 Puppr-DEV[22856:10056239] 5.15.0 - [Firebase/Messaging][I-FCM004003] Successfully subscribed to topic /topics/timedSales
```

In this case, the `device` param values are remaining consistent across multiple app starts.

Let me know if there are any other logs you'd like me to provide. In the meantime, I'll be downgrading to v5.15.0 to work around the issue.

**Comment by chliangGoogle** [comment]
Looks like the parameters are consistent, I will forward the request to the team and get back to you soon.

**Comment by long1eu** [comment]
I also got this. Why is it not fixed?

**Comment by iPhoNewsRO** [comment]
i'm having the same issue, downgrading to v5.15.0 worked

```
pod 'Firebase/Core', '~> 5.15.0'
pod 'Firebase/Messaging', '~> 5.15.0'
```

**Comment by chliangGoogle** [comment]
Notice your device ID has changed, this should be the same issue that one of our previous release cause your client to have a bad checkin. https://github.com/firebase/firebase-ios-sdk/pull/2860 should fix the issue, and it will be in the next release.

In the meantime, you can either patch #2860 or call deleteIDWithHandler: once(and only once) to clear your corrupted checkin data.
